docs/contributing.md

```diff
@@ -0,0 +1,172 @@
+- **Extend the Qualcomm Linux ecosystem:** encourage community participation and long-term maintainability aligned with `meta-qcom`.
+
+References:
+- [Yocto Project Overview](https://docs.yoctoproject.org/overview-manual/yp-intro.html)
+- [OpenEmbedded Layer Index](https://layers.openembedded.org/layerindex/)
+
+---
+
+## 2  General Contribution Guidelines
+
+Our process mirrors the official Yocto Project contribution flow — see
+[Preparing Changes for Submission](https://docs.yoctoproject.org/dev/contributor-guide/submit-changes.html#preparing-changes-for-submission).
+
+### 2.1  Pull-Request Workflow
+
+- **Fork and propose changes** via GitHub Pull Requests.
+  Use **draft mode** for work-in-progress patches.
+- **Create clean commits:** one logical change per commit.
+  Follow [Yocto commit style](https://docs.yoctoproject.org/dev/contributor-guide/submit-changes.html#writing-good-commit-messages).
+- **Explain _why_** the change is needed in the commit message.
+- **Add a Signed-off-by line** to certify compliance with the [Developer’s Certificate of Origin](https://developercertificate.org/).
+- **Validate locally** before submission: build with `bitbake`, flash, and verify runtime.
+- **Address review feedback** and re-push to update your PR.
+  Use `git rebase -i` to squash or reorder commits as needed.
+
+### 2.2  Machine-Specific Isolation
+
+Because this layer expects to host multiple vendor platforms:
+- Use **machine overrides** (`:machine` or `:append:machine`) to confine board-specific logic.
+- Avoid cross-contamination between machines or with upstream `meta-qcom`.
+- Do not introduce SoC-generic behavior under a machine-specific path.
+
+Reference: [BitBake Overrides](https://docs.yoctoproject.org/ref-manual/variables.html#var-OVERRIDES)
+
+### 2.3  Repository Organization
+
+All vendor boards live together in a single layer:
+- **No branch or folder segregation per vendor.**
+- Maintain quality equivalent to `meta-qcom`.
+
+### 2.4  No Recipe Forks
+
+- Forks of recipes from `meta-qcom`, `meta-qcom-hwe`, or base OE / Yocto layers are **not accepted**.
+- Use `.bbappend` files for vendor-specific patching.
+- Keep upstream recipes authoritative.
+
+Reference: [Understanding bbappends](https://docs.yoctoproject.org/ref-manual/terms.html#term-Append-Files)
+
+### 2.5  Scope of Changes
+
+- Limit changes to **BSP-specific content** (kernel, firmware, device tree, drivers, partition configs).
+- Avoid distribution-specific logic — vendors may ship separate distro layers.
+
+Preferred test distros:
+- [`poky-altcfg`](https://git.yoctoproject.org/poky/tree/meta-poky/conf/distro/poky-altcfg.conf) (systemd-compatible)
+- [`meta-qcom-distro`](https://github.com/qualcomm-linux/meta-qcom-distro)
+
+### 2.6  Maintainer Expectations
+
+- Each contributor acts as the **maintainer** of their changes, upstream and downstream.
+- Vendors must appoint a **point-of-contact (PoC)** to review and triage vendor-specific PRs and issues promptly.
```

Comment:
What does it mean exactly?  The PoC's vendor is the person to merge the PR at the end? Is there a mantainers.md file with the list of people acting as maintainer and the person submitting the PR must include that person as reviewer?
I think elaboration is needed in this description

Reply:
Yeah, a maintainers.md file would work ok here.